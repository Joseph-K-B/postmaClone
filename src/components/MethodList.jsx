import React from "react";
import Method from "./app/Method";


export default async function MethodList({ methodList }) {
    const methods = await methodList.map(({method, url}) => {
        <li>
            <Method 
            method={method}
            url={url}
            />
        </li>
    })
    if(methodList === undefined) {
        return <p>Waiting for request</p>
    } else {
    return <ul aria-label='method-list'>{methods}</ul>}
}