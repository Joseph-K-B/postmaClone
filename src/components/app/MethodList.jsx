import React from "react";
import Method from "./Method";


export default function MethodList({ methodList, contents }) { 
    if(contents === undefined) {
        return <p>Waiting for request</p>
    } else {
        const methods = methodList.push(({method, url}) => {
            <li>
                <Method 
                method={method}
                url={url}
                />
            </li>
        })
    return <ul aria-label='method-list'>{methods}</ul>
    }
}