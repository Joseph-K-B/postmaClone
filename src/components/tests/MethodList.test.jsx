import React from 'react';
import { render } from '@testing-library/react';
import MethodList from '../app/MethodList';

describe('renders list of API calls made', () => {
    it('should display initial message of please make request', () => {
        render(
            <MethodList />
        );
        expect(MethodList).toMatchSnapshot();
    });
});