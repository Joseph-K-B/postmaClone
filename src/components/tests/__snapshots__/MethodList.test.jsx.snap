// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of API calls made should display initial message of please make request 1`] = `[Function]`;
